.github/workflows/beaker.yml

```diff
@@ -176,6 +181,7 @@ jobs:
           bundler-cache: true
           cache-version: ${{ inputs.cache-version }}
           working-directory: ${{ inputs.working-directory }}
+          self-hosted: ${{ inputs.acceptance_runs_on != 'ubuntu-24.04' }}
```

Comment:
Shouldn't the auto-detection in https://github.com/ruby/setup-ruby/blob/a6e6f86333f0a2523ece813039b8b4be04560854/common.js#L165-L176 work for that already?

Reply:
one would think! Basically: If the OS of the self-hosted runner doesn't look like the OS GitHub uses, auto-detection works. The CERN-provided runners are on EL9, so that works. Now this time I wanted to be smart and thought 'it is easier to configure the runner software if my OS is the same as GitHub uses', so our new runners are... Ubuntu 24.04. And that breaks auto-detection :sadface:

Reply:
:woman_facepalming: 

okay!

Reply:
my problem with this is that if anyone decides to set `acc_runs_on` to *anything* but `24.04`, it will trigger, even if it's not really a self-hosted runner.

Reply:
```suggestion
          self-hosted: ${{ ! startsWith(inputs.acceptance_runs_on, 'ubuntu-') }}
```

Reply:
will test it and provide a PR if it works

Reply:
works: https://github.com/voxpupuli/gha-puppet/pull/64